web/src/stream_data.ts

```diff
@@ -720,7 +740,7 @@ export function clean_up_description(sub) {
     }
 }
 
-export function create_sub_from_server_data(attrs) {
+export function create_sub_from_server_data(attrs: StreamAttrs): StreamSubscription {
     if (!attrs.stream_id) {
         // fail fast
         throw new Error("We cannot create a sub without a stream_id");
```

Comment:
Down below there is a line `delete attrs.subscribers`. If we avoid that, `subscribers` do not need to be optional.

Reply:
I think `subscribers` should be optional here because the type for `ApiStreamSubscription` does in fact have `subscribers` as an optional property.
https://github.com/zulip/zulip/blob/6632eca2dcf2051043e95348658d6da02305669f/zerver/lib/types.py#L198.

Also we are not setting it to a default of maybe `[]` while making `attrs` for `create_sub_from_server_data` here - 
https://github.com/zulip/zulip/blob/6632eca2dcf2051043e95348658d6da02305669f/web/src/stream_data.js#L834-L841

Reply:
Hmm, that makes sense. Since we are deleting it from `attrs` before expanding it to `sub`, do we want to remove `subscribers` from `StreamSubscription`?

Reply:
Yeah it makes sense to remove it from `StreamSubscription` 👍